### PR TITLE
Post test result summary in workflow run and in PR comment using github-actions[bot]

### PR DESCRIPTION
**Pull request by rodelgc**
### Changes proposed in this Pull Request:

This PR posts a summary of test results in the following places:

1. In the workflow run, like so:
  Screen Shot 2022-07-04 at 8 08 31 PM

2. In a PR comment, effectively replacing the comment that `botwoo` creates. This time, the new PR comment is by `github-actions[bot]`.
  Screen Shot 2022-07-04 at 8 08 46 PM

This will work on both Playwright and Puppeteer E2E tests.

Closes https://github.com/woocommerce/woocommerce-quality/issues/305.

### How to test the changes in this Pull Request:

1. Duplicate this PR.
2. In the duplicate PR, toggle the `E2E_PLAYWRIGHT` flag on by setting this line of code to `${{ true }}` in the `pr-build-and-e2e-tests.yml` workflow file:
  https://github.com/woocommerce/woocommerce/blob/3e6196a31229bbb02ddd94acfab85ad43cbf47e5/.github/workflows/pr-build-and-e2e-tests.yml#L11
3. Wait for the new "Post test results" job to finish.
    Screen Shot 2022-07-04 at 8 11 28 PM
4. Verify that you should see the "Test Results Summary" at the end of the workflow run page, like so:
    Screenshot 2022-07-04 at 20-30-58 Post test result summary in workflow run and in PR comment using github-actions bot · woocommerce_woocommerce@e38962a
5. Go to the PR page and verify that you no longer see `botwoo`'s PR comment. Instead, you would see `github-actions[bot]`'s test results summary like so:
    Screen Shot 2022-07-04 at 8 08 46 PM
6. Switch the `E2E_PLAYWRIGHT` toggle to `false` so that this time, the Puppeteer e2e tests would run.
7. Again, wait for the "Post test results" job to finish.
8. Again, verify that you see the "Test Results Summary" at the end of this Puppeteer workflow run.
9. When you go back to the PR page in GitHub, notice that the PR comment by `github-actions[bot]` earlier was updated -- it's now showing the new Commit SHA and the new values in the `E2E Tests` row.
10. Verify that the links in the test summary are working. The link to the full E2E test report points to the WooCommerce Test Reports Dashboard for the meantime.

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: .github/workflows/pr-build-and-e2e-tests.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
 
 env:
-  E2E_PLAYWRIGHT: ${{ false }}
+    E2E_PLAYWRIGHT: ${{ false }}
```
Comment (rodelgc) [inline .github/workflows/pr-build-and-e2e-tests.yml:11]: AFAIK we're using 4 spaces in the default style guide for YAML files, so I think leaving these changes in spacing is OK?

**Comment by rodelgc** [comment]
I just noticed that the test summary looks a bit awful on the email notification. Oh well 🤷‍♂️ 
Screen Shot 2022-07-04 at 9 27 01 PM

File: .github/workflows/pr-build-and-e2e-tests.yml (hunk 1 above)
Comment (zhongruige) [reply .github/workflows/pr-build-and-e2e-tests.yml:11]: > we're using 4 spaces in the default style guide for YAML files

Good question, I am not sure if we're using a set style, but the file is still valid YAML and works as expected, so I think this should be OK as well.

**Review by zhongruige — APPROVED** [review]
Thank you for the great work on this @rodelgc! Changes look good to me, and I like how the comment is formatted--especially the new table.

**Pull request merged**